typewise_alert.test.py

```diff
@@ -1,11 +1,130 @@
 import unittest
+from alerts import ControllerAlert, EmailAlert
+from config_parser import ConfigProvider
 import typewise_alert
 
 
 class TypewiseTest(unittest.TestCase):
-  def test_infers_breach_as_per_limits(self):
-    self.assertTrue(typewise_alert.infer_breach(20, 50, 100) == 'TOO_LOW')
+    def test_classify_temperature_breach_for_passive_cooling_with_breach_type_normal(
+        self,
+    ):
+        self.assertTrue(
+            typewise_alert.classify_temperature_breach("PASSIVE_COOLING", 20)
+            == "NORMAL"
+        )
 
+    def test_classify_temperature_breach_for_passive_cooling_with_breach_type_too_low(
+        self,
+    ):
+        self.assertTrue(
+            typewise_alert.classify_temperature_breach("PASSIVE_COOLING", -1)
+            == "TOO_LOW"
+        )
 
-if __name__ == '__main__':
-  unittest.main()
+    def test_classify_temperature_breach_for_passive_cooling_with_breach_type_too_high(
+        self,
+    ):
+        self.assertTrue(
+            typewise_alert.classify_temperature_breach("PASSIVE_COOLING", 100)
+            == "TOO_HIGH"
+        )
+
+    def test_classify_temperature_breach_for_hi_active_cooling_with_breach_type_normal(
+        self,
+    ):
+        self.assertTrue(
+            typewise_alert.classify_temperature_breach("HI_ACTIVE_COOLING", 20)
+            == "NORMAL"
+        )
+
+    def test_classify_temperature_breach_for_hi_active_cooling_with_breach_type_too_low(
+        self,
+    ):
+        self.assertTrue(
+            typewise_alert.classify_temperature_breach("HI_ACTIVE_COOLING", -1)
+            == "TOO_LOW"
+        )
+
+    def test_classify_temperature_breach_for_hi_active_cooling_with_breach_type_too_high(
+        self,
+    ):
+        self.assertTrue(
+            typewise_alert.classify_temperature_breach("HI_ACTIVE_COOLING", 46)
+            == "TOO_HIGH"
+        )
+
+    def test_classify_temperature_breach_for_med_active_cooling_with_breach_type_normal(
+        self,
+    ):
+        self.assertTrue(
+            typewise_alert.classify_temperature_breach("MED_ACTIVE_COOLING", 20)
+            == "NORMAL"
+        )
+
+    def test_classify_temperature_breach_for_med_active_cooling_with_breach_type_too_low(
+        self,
+    ):
+        self.assertTrue(
+            typewise_alert.classify_temperature_breach("MED_ACTIVE_COOLING", -1)
+            == "TOO_LOW"
+        )
+
+    def test_classify_temperature_breach_for_med_active_cooling_with_breach_type_too_high(
+        self,
+    ):
+        self.assertTrue(
+            typewise_alert.classify_temperature_breach("MED_ACTIVE_COOLING", 41)
+            == "TOO_HIGH"
+        )
+
+    def test_email_check_and_alert_for_hi_active_cooling_with_breach_type_too_high(
```

Comment:
There needs to be test cases that verify if the right alert target has received the intended message